src/ts/components/timelinemarkers.ts

```diff
@@ -0,0 +1,208 @@
+
+    if (this.timelineMarkers.length > 0) {
+      for (let marker of this.timelineMarkers) {
+        // Handle interval markers
+        if (marker.duration > 0
+          && percentage >= marker.position - snappingRange
+          && percentage <= marker.position + marker.duration + snappingRange) {
+          return marker;
+        }
+        // Handle position markers
+        else if (percentage >= marker.position - snappingRange
+          && percentage <= marker.position + snappingRange) {
+          return marker;
+        }
+      }
+    }
+
+    return null;
+  }
+
+  private clearMarkers = (): void  => {
+    this.timelineMarkers = [];
+    this.markersContainer.empty();
+  }
+
+  private removeMarker(marker: TimelineMarker): void {
+    this.uimanager.getConfig().metadata.markers = this.uimanager.getConfig().metadata.markers.filter(_marker => marker !== _marker);
+  }
+
+  private updateMarkers(): void {
+    this.clearMarkers();
+
+    if (!shouldProcessMarkers(this.player, this.uimanager)) {
+      return;
+    }
+
+    for (let marker of this.uimanager.getConfig().metadata.markers) {
+      const { markerPosition, markerDuration } = getMarkerPositions(this.player, marker);
+
+      if (shouldRemoveMarker(markerPosition, markerDuration)) {
+        this.removeMarker(marker);
+      } else {
+        this.timelineMarkers.push({ marker, position: markerPosition, duration: markerDuration });
+      }
+    }
+
+    // Populate the timeline with the markers
+    this.updateMarkersDOM();
+  }
+
+  private updateMarkersDOM(): void {
+    // TODO: For live streams we are removing all markers, and re-adding them.
+    // we should track markers by some internal IDs and just update their positions
+    // This would give us ability to make their pos updating smooth with css transition
+    this.markersContainer.empty();
+
+    const seekBarWidthPx = this.getSeekBarWidth();
+
+    for (let marker of this.timelineMarkers) {
+      const markerClasses = ['seekbar-marker'].concat(marker.marker.cssClasses || [])
+        .map(cssClass => this.prefixCss(cssClass));
+
+      const cssProperties: {[propertyName: string]: string} = {
+        'width': marker.position + '%',
+      };
+
+      if (marker.duration > 0) {
+        const markerWidthPx = Math.round(seekBarWidthPx / 100 * marker.duration);
+        cssProperties['border-right-width'] = markerWidthPx + 'px';
+        cssProperties['margin-left'] = '0';
+      }
+
+      this.markersContainer.append(new DOM('div', {
+        'class': markerClasses.join(' '),
+        'data-marker-time': String(marker.marker.time),
+        'data-marker-title': String(marker.marker.title),
+      }).css(cssProperties));
+    }
+  }
+
+  private configureLivePausedTimeshiftUpdater(
+    handler: () => void,
+  ): void {
+    // Regularly update the marker position while the timeout is active
+    this.pausedTimeshiftUpdater = new Timeout(1000, handler, true);
```

Comment:
Do we really need two of those? 🤔 One here and the very same is still in the `seekbar.ts`

Reply:
Indeed, did not want to pass the method in as a parameter. I could extract its functionality outside and re-use the method?

Reply:
We could introduce another helper class which is handling this. 🤔 

What about adding a `PausedTimeshiftUpdatedHandler` where you can attach listeners. So the seekbar and this component would attach a listener 🤔 That would result in just having one `Timeout`.

But not sure if this isn't an overkill here 😄

Reply:
Little overkill, but sure, I will make it as a `singleton` class.

Reply:
Introduced in https://github.com/bitmovin/bitmovin-player-ui/pull/310/commits/2f59f252295301384efae0529a6e626050f6e5c5 and https://github.com/bitmovin/bitmovin-player-ui/pull/310/commits/c82874e6b323c0b4d7e2fb87e1ddc8c299e9e9c4